tests/unit/test_util.py

```diff
@@ -187,6 +189,50 @@ def test_solve_qp_invalid_gramian_row_mean(self) -> None:
                 gramian_row_mean="invalid_gramian_row_mean",
             )
 
+    @pytest.mark.parametrize(
+        "data_size, batch_size, num_batches",
+        [
+            (1.0, 1, 1),
+            (1, 1.0, 1),
+            (1, 1, 1.0),
+            (-1, 1, 1),
+            (1, -1, 1),
+            (1, 1, -1),
+            (0, 1, 1),
+            (1, 0, 1),
+            (1, 1, 0),
+            (1, 2, 1),
+        ],
+        ids=[
+            "float_data_size",
+            "float_batch_size",
+            "float_num_batches",
+            "negative_data_size",
+            "negative_batch_size",
+            "negative_num_batches",
+            "zero_data_size",
+            "zero_batch_size",
+            "zero_num_batches",
+            "data_size_smaller_than_batch_size",
+        ],
+    )
+    def test_sample_batch_indices(
+        self,
+        data_size: int,
+        batch_size: int,
+        num_batches: int,
+    ) -> None:
+        """
+        Test sample_batch_indices for valid input parameters.
+        """
+        with pytest.raises(ValueError):
```

Comment:
This test should also check the desired/required properties of the result (uniqueness of the indices, etc...)

Reply:
Added uniqueness test